### ISSUE TITLE
Improper html link generation, leaving last colon as text.

### ISSUE THREAD

**Issue by pkgodara** [gregjacobs/Autolinker.js#206]
If we type "a.:ok:" then it gives link to the text "a.:ok" and last ':' is printed as plain text.
other ex:
"a.:t1:t2:" gives "a.:t1:t2" and ":"

**Comment by gregjacobs** [comment]
Hey @pkgodara, can you give a more detailed example / real world use case?

**Comment by pkgodara** [comment]
@gregjacobs  I'm working on a messenger which uses colon (':') for special purposes like emoji.
Ex  "hi :smile:" should produce "hi <smile emoji here>".
Now if I type "hi link.:smile::cat:" then it should give link to "link.:smile::cat:" , but it separates last colon and hence last emoji will not be shown in that case.

**Comment by gregjacobs** [comment]
Hey @pkgodara. Hmm, sorry, but I'm still not following. What is "link."? Are the emoji's part of a URL?

**Comment by gregjacobs** [comment]
Perhaps autolinker shouldn't even be linking that string "hi link.:emoji1::emoji2:"?

**Comment by pkgodara** [comment]
Hello @gregjacobs ,Let me give an example ,
I typed: "Hello test.:emoji1::emoji2: text here.", the output shows: "Hello **test.:emoji1::emoji2**: text here." where bold is a html link. 
The console output which I printed is
"Hello <a href="test.:emoji1::emoji2" target="_blank" rel="noopener noreferrer">test.:emoji1::emoji2</a>: text here."
The url regex used is "(://|www\.).+" .

**Comment by pkgodara** [comment]
And emoji's are part of URL.

**Comment by simison** [comment]
@pkgodara do you mean that instead of word "test" there is an url like so: `http://www.example.com/:emoji1::emoji2:` ? Sorry but those link/test words are a bit confusing.

**Comment by pkgodara** [comment]
@simison Yes, test is an url. And sometimes this creates problems in rendering emoji as last colon is separated.
